### ISSUE TITLE
Input Date Picker - Ensure custom controls provide proper textual name, role, and state information - (2036569971)

### ISSUE THREAD

**Issue by dqateam** [Esri/calcite-components#5583]
### Module: 
 17a Input Date Picker 

### Violation: 
 Ensure custom controls provide proper textual name, role, and state information 

![image](https://user-images.githubusercontent.com/5023024/202756234-6dca9d9e-7fdc-4f5b-aabe-2dd6694756c7.png)

### WCAG Reference: 
- [1.1.1 Non-text Content](https://www.w3.org/WAI/WCAG21/Understanding/non-text-content.html)
- [4.1.2 Name, Role, Value](https://www.w3.org/WAI/WCAG21/Understanding/name-role-value.html) 

### Instance ID: 
 2036569971 

### Severity: 
 7 

### Description: 
 [Issue]
The calendar "dates" do have role and state information. Currently it is announcing as a plain text to screen reader user.

[User Impact]
Screen reader users will have difficulty determining the purpose and state of these controls.

[Code Reference] 
 ```
<div class="day-v-wrapper">
<div class="day-wrapper">
<span class="day"><span class="text">18</span></span>
</div>
</div>
 ``` 

### Note: 
 [Suggestion]
Ensure custom controls provide proper textual name, role, and state information. Properly structured native controls provide this information automatically. For custom controls, developers must explicitly add this information by using ARIA attributes.

Most modern browsers have an Accessibility section in the developer tools panel that will display an element&#039;s calculated name, role, and state information.

A control&#039;s role indicates the type of control, such as "checkbox" or "button". For custom controls, the role must be set with the role attribute. The ARIA specification only permits certain values for this attribute. Additionally, the ARIA in HTML specification does prevents the use of certain roles on some elements. 

A control&#039;s state indicates several possible pieces of information. Common states include expanded, current, selected, checked, and disabled. Most are indicated with similarly named ARIA attributes. Certain states can only be used with certain roles.

[Compliant Code Example] 
 ```
N/A




 ```
[Recommended Reading]
For more information about creating accessible custom date pickers, please see the WhatSock website: https://whatsock.com/Templates/Datepickers/index.htm 

### Media Type: 
 Custom Controls 

### Additional Resources: 
- Report Source: Product accessibility evaluation conducted on the Calcite Design System in August, 2022.

**Comment by geospatialem** [comment]
We're fulfilling some of the basic requirements, but perhaps we could add a bit more context to the date?

Currently once in the date area screen readers are provided with the number ("1", but no other context. Perhaps could provide a bit more context to the month and year?

**Comment by geospatialem** [comment]
Duplicate of #5570.
